### PR TITLE
split summary.csv into summary and summary_extended

### LINKED ISSUE #856: Two different results report files, one with ALL details and one with a summary of the data
We’ve been adding more granular inactive ballot buckets to help with reporting. These granular buckets are important for reporting on RCV contests, but are starting to take a lot of real estate and grab maybe too much attention in the `summary.csv`. So we’re proposing to split the current `summary.csv` into two separate files…

### summary_report.csv
Will include most of the 2.0 updates like round by round vote percentages and round by round transfers but will NOT include the split out inactive buckets. It will have just one "Inactive Ballots Total" row similar to all output before 2.0.
- Number of undervotes in round 0
- For each candidate, each round
  - % of total votes for each round
  - Number of ballots transferred from eliminated candidate(s) in the previous round
- ~Inactive ballots in each round broken down by category~
  - ~Overvote~
  - ~Skipped Ranking~
  - ~Repeat Ranking~
  - ~Exhausted Choices~
- Per-Round winning threshold
- RCTab version number
- Contest Summary
- Number to be Elected
- Number of Candidates
- Total Number of Ballots

### detailed_report.csv 

will include all of the current 2.0 updates.

### detailed_report.json
`summary.json` will include everything and be renamed `detailed_report.json`.

### Tabulate by Slice
 Tabulate by slice files will include everything and be renamed `[sliceId]_[sliceType]_detailed_report.[csv|json]`

### PR DESCRIPTION
**Pull request by artoonie**
Closes #856 

Adds tests for every summary.csv file without bloating the repo by including that mostly-duplicate data.

### CODE REVIEW

File: src/test/java/network/brightspots/rcv/TabulatorTests.java [hunk 1]
```diff
@@ -365,6 +368,54 @@ private static boolean compareFiles(
...
+        if (lineExtended.startsWith("Inactive Ballots by")) {
+          continue;
+        }
+
+        // This line should be equal in both files. Ensure the line exists and they're equal
+        // in both files.
+        String lineSummary = brSummary.readLine();
+        assertNotNull(lineSummary, "Summary file is missing a line");
+        if (!lineSummary.equals(lineExtended)) {
+          fail();
```
Comment (yezr) [inline src/test/java/network/brightspots/rcv/TabulatorTests.java:407]: So I made a change to force a mismatch between the summary and extended_summary file text. It caused this line to fail as expected, but it was just a generic stack trace that was hard to hunt down. Is it possible to change this to include a more human readable failure message like "Line X mismatch between summary and extended_summary"?

**Comment by yezr** [comment]
### Testing
- Made a summary.csv and extended_summary.csv with different values and confirmed that the new test failed

**Comment by yezr** [comment]
added a message in #856 about changing the tab by slice summary names to prevent the confusion @artoonie raised in the same issue

File: src/test/java/network/brightspots/rcv/TabulatorTests.java (hunk 1 above)
Comment (artoonie) [reply src/test/java/network/brightspots/rcv/TabulatorTests.java]: done!

File: src/main/java/network/brightspots/rcv/ResultsWriter.java [hunk 2]
```diff
@@ -280,26 +280,48 @@ void generateBySliceSummaryFiles(
...
         String sliceFileString = getFileStringForSlice(slice, sliceId, filenames);
-        String outputPath = getOutputFilePathFromInstance(
-            String.format("%s_summary", sliceFileString));
-        generateSummarySpreadsheet(roundTallies, candidateOrder, slice, sliceId, outputPath);
-        generateSummaryJson(roundTallies, tallyTransfers, slice, sliceId, outputPath);
+        String outputPathCsv = getOutputFilePathFromInstance(
+            String.format("%s_extended_summary", sliceFileString));
+        String outputPathJson = getOutputFilePathFromInstance(
+                String.format("%s_summary", sliceFileString));
+        generateSummaryExtendedCsv(roundTallies, candidateOrder, slice, sliceId, outputPathCsv);
```
Comment (artoonie) [inline src/main/java/network/brightspots/rcv/ResultsWriter.java:287]: Make `summaryExtended` versus `extended_summary` order consistent

File: src/main/java/network/brightspots/rcv/ResultsWriter.java [hunk 3]
```diff
@@ -280,26 +280,48 @@ void generateBySliceSummaryFiles(
         RoundTallies roundTallies = entry.getValue();
         TallyTransfers tallyTransfers = tallyTransfersBySlice.get(slice, sliceId);
         String sliceFileString = getFileStringForSlice(slice, sliceId, filenames);
-        String outputPath = getOutputFilePathFromInstance(
-            String.format("%s_summary", sliceFileString));
-        generateSummarySpreadsheet(roundTallies, candidateOrder, slice, sliceId, outputPath);
-        generateSummaryJson(roundTallies, tallyTransfers, slice, sliceId, outputPath);
+        String outputPathCsv = getOutputFilePathFromInstance(
+            String.format("%s_extended_summary", sliceFileString));
```
Comment (artoonie) [inline src/main/java/network/brightspots/rcv/ResultsWriter.java:284]: this string format is repeated a few times-- consider refactoring

**Review by yezr — APPROVED** [review]
big ups to @alyssahursh, @artoonie and @nurse-the-code  for all the work on this. LGTM!

**Pull request merged**